worlds/pokemon_crystal/items.py

```diff
@@ -49,20 +50,65 @@ def item_const_name_to_id(const_name):
     if ids:
         return ids[0]
     return 0
-
-
-def get_random_filler_item(random: Random):
-    # weights are roughly based on vanilla occurrence
-    weighted_pool = [["RARE_CANDY"] * 3, ["ETHER", "ELIXER", "MAX_ETHER", "MAX_ELIXER", "MYSTERYBERRY"] * 5,
-                     ["WATER_STONE", "FIRE_STONE", "THUNDERSTONE", "LEAF_STONE", "SUN_STONE", "MOON_STONE"] * 2,
-                     ["ESCAPE_ROPE"] * 3, ["NUGGET", "STAR_PIECE", "STARDUST", "PEARL", "BIG_PEARL"] * 2,
-                     ["POKE_BALL", "GREAT_BALL", "ULTRA_BALL"] * 5,
-                     ["POTION", "SUPER_POTION", "ENERGY_ROOT", "ENERGYPOWDER"] * 12,
-                     ["HYPER_POTION", "FULL_RESTORE"] * 2, ["REPEL", "SUPER_REPEL", "MAX_REPEL"] * 3,
-                     ["REVIVE", "REVIVAL_HERB"] * 4 + ["MAX_REVIVE"] * 2,
-                     ["HP_UP", "PP_UP", "PROTEIN", "CARBOS", "CALCIUM", "IRON"] * 5,
-                     ["GUARD_SPEC", "DIRE_HIT", "X_ATTACK", "X_DEFEND", "X_SPEED", "X_SPECIAL"] * 2,
-                     ["HEAL_POWDER", "BURN_HEAL", "PARLYZ_HEAL", "ICE_HEAL", "ANTIDOTE", "AWAKENING", "FULL_HEAL"] * 5]
+    
+def get_fill_one(world: "PokemonCrystalWorld"):
+    if world.options.item_pool_fill == ItemPoolFill.option_balanced:
+        return "balanced"
+    elif world.options.item_pool_fill == ItemPoolFill.option_youngster:
+        return "youngster"
+    elif world.options.item_pool_fill == ItemPoolFill.option_vanilla:
+        return "vanilla"
+    elif world.options.item_pool_fill == ItemPoolFill.option_cooltrainer:
+        return "cooltrainer"
+    else:
+        return "oops"
+    
+
+def get_random_filler_item(random: Random, world: "PokemonCrystalWorld"):
```

Comment:
```suggestion
def get_random_filler_item(world: "PokemonCrystalWorld"):
```
Since the `World` instance is being passed here now there's no need to pass `random` separately anymore